src/Account.sol

```diff
@@ -845,6 +919,11 @@ contract Account is IAccount, Auth, OpsReady {
         ConditionalOrder memory conditionalOrder =
             getConditionalOrder(_conditionalOrderId);
 
+        /// @dev return false early if market key is the default value (i.e. "")
+        if (conditionalOrder.marketKey == bytes32(0)) {
+            return false;
+        }
```

Comment:
**Question/suggestion:** I wonder if it is worth having a similar check within `_getPerpsV2Market`, which as far as I can tell currently, will return `IPerpsV2MarketConsolidated(address(0))` if the `_marketKey` is invalid. Although in theory this would not cause any downstream errors in the example of `_validConditionalOrder` due to a downstream failure in the `_sUSDRate` function I think it would make sense if this function had the following the check, on basis of the principal of "fail fast, fail early".
```solidity
    function _getPerpsV2Market(bytes32 _marketKey)
        internal
        view
        returns (IPerpsV2MarketConsolidated)
    {
        address market = FUTURES_MARKET_MANAGER.marketForKey(_marketKey);
        if (market == address(0)) revert InvalidMarketKey(_marketKey);
        return IPerpsV2MarketConsolidated(market);
    }
```
This is not a critical issue for the reasons mentioned, but I feel that this function failing is what would be expected if it is provided an invalid `_marketKey`. Furthermore the downstream failure is based on upgradeable code that we don't control, so better to program defensively I think just incase of a wildcard scenario.

Reply:
Added a simple assertion. A custom error is just adding more code after the audit at this point and likely will *never* be thrown.

And _sUSDRate can only be reached after a successful internal call to _getPerpsV2Market which will revert if the market address returned is address(0) (after I add assertion)

Reply:
Added assertion check ✅